### PR TITLE
Survey v2

### PR DESCRIPTION
**Pull request by ksvitkovsky**

### CODE REVIEW

**Comment by jsedlacek** [comment]
@ksvitkovsky Can we have a bit bigger gap between rating options, just like our in-app survey has? The current seems too crowded.

Snímek obrazovky 2019-11-28 v 10 34 57
Snímek obrazovky 2019-11-28 v 10 34 51

**Comment by jsedlacek** [comment]
@ksvitkovsky Is the new version responsive? How can I test that?

**Comment by jsedlacek** [comment]
@ksvitkovsky In the new version the rating does not seem to be completely aligned to border.
 
Snímek obrazovky 2019-11-28 v 10 41 08

Snímek obrazovky 2019-11-28 v 10 40 31

**Comment by ksvitkovsky** [comment]
@jsedlacek yes, it's the same as with other web pages - you can simply resize the window or open preview in device size (open dev tools and then cmd + shift + m)

**Comment by jsedlacek** [comment]
> @jsedlacek yes, it's the same as with other web pages - you can simply resize the window or open preview in device size (open dev tools and then cmd + shift + m)

Thanks. Lgtm. It seems the breakpoints are different, so I did not see it when I just resized the window.

**Comment by jsedlacek** [comment]
@ksvitkovsky It seems the whole button is not "active" (clicking on it won't work), just the middle.

Nov-28-2019 10-52-04

Snímek obrazovky 2019-11-28 v 10 51 19

File: src/transformV2.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    right: direction === 'ltr' ? 'right' : 'left',
+    unsubscribeUrl: options.unsubscribeUrl,
+    preview: is.boolean(options.preview) ? options.preview! : false,
+    showPoweredBy: is.boolean(options.showPoweredBy)
+      ? options.showPoweredBy!
+      : true,
+    width: NPS_WIDTH
+  } as any;
+
+  if (options.choices) {
```
Comment (jsedlacek) [inline src/transformV2.ts]: How do we distinguish which survey type to show? Just based on `choices` etc? Can we rather pass question type, just like we do in `satismeter-survey`?

File: src/partials/number-scale.hbs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+              <tbody>
+                <tr>
+                  <td style="vertical-align:top;padding:1px;">
+                    <table border="0" cellpadding="0" cellspacing="0" role="presentation" style="" width="100%">
+                      <tr>
+                        <td align="center" vertical-align="middle" style="font-size:0px;padding:0px;word-break:break-word;">
+                          <table border="0" cellpadding="0" cellspacing="0" role="presentation" style="border-collapse:separate;width:100%;line-height:100%;">
+                            <tr>
+                              <td align="center" bgcolor="{{../colors.primary}}" role="presentation" style="border:none;border-radius:3px;cursor:auto;height:40px;mso-padding-alt:0px 10px;background:{{../colors.primary}};"
+                                valign="middle"> <a {{#if url}} href="{{{url}}}" {{/if}} style="display:inline-block;background:{{../colors.primary}};color:{{../colors.background}};font-family:airal, helvetica, sans-serif;font-size:14px;font-weight:normal;line-height:120%;margin:0;text-decoration:none;text-transform:none;padding:10px 10px;mso-padding-alt:0px;border-radius:3px;"
```
Comment (jsedlacek) [inline src/partials/number-scale.hbs]: Seems typo: `airal`

**Comment by jsedlacek** [comment]
@ksvitkovsky We don't pass the value of rating in examples.

Snímek obrazovky 2019-11-28 v 11 02 02

File: src/transformV2.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
+import is from 'is';
+import Uri from 'jsuri';
+import marked from 'marked';
+
+import { BaseTemplateOptions, Colors, Rating, TemplateV2Options } from './base';
+import messages from './messages';
+
+export type TransformV2Options = {
+  choices?: string[];
```
Comment (jsedlacek) [inline src/transformV2.ts]: The definition combines unrelated fields from multiple question types. Can we have some nice interface, that will make clear what properties are related to question (interface per question type) and which fields are related to the email itself?

**Comment by ksvitkovsky** [comment]
WIP

**Comment by jsedlacek** [comment]
@ksvitkovsky any chance we can have storybook here? It's much nicer to review examples, but I am not sure how would it work with our handlebars templates.

**Comment by ksvitkovsky** [comment]
@jsedlacek storybook added, had to update our setup with webpack

**Comment by jsedlacek** [comment]
@ksvitkovsky Good job! Released as `@satismeter/nps-email@2.0.0`.

**Pull request merged**